lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -22,11 +24,10 @@ def generate_additional_resources(resource)
     generated = [generated] unless generated.is_a?(Array)
     generated.collect do |res|
       @catalog.resource(res.ref) || res
-    end.each do |res|
-      priority = @prioritizer.generate_priority_contained_in(resource, res)
-      add_resource(res, resource, priority)
+    end.reverse_each do |res|
```

Comment:
Before, when stuff was being added to the evaluation graph, the new resources were given sequential priorities just after the parent resource.

Now they're being added to the catalog in sequence, each one immediately after the parent resource. This causes them to be reversed. Keeping their order would involve making the "Add this resource to the catalog" step way more complicated.

Order doesn't matter for any built-in resources, but I didn't want to change behavior